broadbean/broadbean.py

```diff
@@ -27,16 +27,34 @@ class PulseAtoms:
 
     @staticmethod
     def sine(freq, ampl, off, phase, SR, npts):
-        time = np.linspace(0, npts/SR, int(npts), endpoint=False)
-        freq *= 2*np.pi
-        return (ampl*np.sin(freq*time+phase)+off)
+        time = np.linspace(0, npts / SR, int(npts), endpoint=False)
+        freq *= 2 * np.pi
+        return ampl * np.sin(freq * time + phase) + off
 
     @staticmethod
     def ramp(start, stop, SR, npts):
-        dur = npts/SR
-        slope = (stop-start)/dur
+        dur = npts / SR
+        slope = (stop - start) / dur
         time = np.linspace(0, dur, int(npts), endpoint=False)
-        return (slope*time+start)
+        return slope * time + start
+
+    @staticmethod
+    def arb_func(func: Callable, kwargs, SR, npts):
```

Comment:
perhaps mention this function in the tutorial notebook to document its existence? https://github.com/picarro-yren/broadbean/blob/yure/arb_function/docs/examples/Pulse_Building_Tutorial.ipynb

Reply:
Thanks! I updated the notebook to include an example of this feature.